premium/usage.go

```diff
@@ -420,6 +458,35 @@ func (u *BatchUpdater) updateUsageWithRetryAndBackoff(ctx context.Context, rows
 		u.logger.Debug().Str("url", u.url).Int("try", retry).Int("max_retries", u.maxRetries).Uint32("rows", rows).Msg("updating usage")
 		queryStartTime := time.Now()
 
+		// If the AWS Marketplace client is set, use it to track usage
+		if u.awsMarketPlaceClient != nil {
```

Comment:
Maybe move this to `reportUsageToAWSMarketplace` as `updateUsageWithRetryAndBackoff` is getting quite long